hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java

```diff
@@ -676,6 +677,60 @@
 
 
     /**
+     * Type of Query Index result copying behavior.
+     *
+     * Defines the behavior for index copying on index read/write.
+     *
+     * Supported in BINARY, OBJECT and NATIVE in-memory-formats.
+     *
+     * Why is it needed? In order to support correctness the internal data-structures used by indexes need to do some copying.
```

Comment:
.......? In order to support the correctness of query results, the underlying data structures used by indexes need to do some copying.